x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/mapper/SemanticTextInferenceResultFieldMapper.java

```diff
@@ -0,0 +1,241 @@
+
+        SemanticTextInferenceFieldType() {
+            super(NAME, true, false, false, TextSearchInfo.NONE, Collections.emptyMap());
+        }
+
+        @Override
+        public String typeName() {
+            return CONTENT_TYPE;
+        }
+
+        @Override
+        public ValueFetcher valueFetcher(SearchExecutionContext context, String format) {
+            return SourceValueFetcher.identity(name(), context, format);
+        }
+
+        @Override
+        public Query termQuery(Object value, SearchExecutionContext context) {
+            return null;
+        }
+    }
+
+    private SemanticTextInferenceResultFieldMapper() {
+        super(SemanticTextInferenceFieldType.INSTANCE);
+    }
+
+    @Override
+    protected void parseCreateField(DocumentParserContext context) throws IOException {
+        XContentParser parser = context.parser();
+        if (parser.currentToken() != XContentParser.Token.START_OBJECT) {
+            throw new DocumentParsingException(parser.getTokenLocation(), "Expected a START_OBJECT, got " + parser.currentToken());
+        }
+
+        parseInferenceResults(context);
+    }
+
+    private static void parseInferenceResults(DocumentParserContext context) throws IOException {
+        XContentParser parser = context.parser();
+        MapperBuilderContext mapperBuilderContext = MapperBuilderContext.root(false, false);
+        for (XContentParser.Token token = parser.nextToken(); token != XContentParser.Token.END_OBJECT; token = parser.nextToken()) {
+            if (token != XContentParser.Token.FIELD_NAME) {
+                throw new DocumentParsingException(parser.getTokenLocation(), "Expected a FIELD_NAME, got " + token);
+            }
+
+            parseFieldInferenceResults(context, mapperBuilderContext);
+        }
+    }
+
+    private static void parseFieldInferenceResults(
+        DocumentParserContext context,
+        MapperBuilderContext mapperBuilderContext) throws IOException {
+
+        String fieldName = context.parser().currentName();
+        Mapper mapper = context.getMapper(fieldName);
+        if (mapper == null || SemanticTextFieldMapper.CONTENT_TYPE.equals(mapper.typeName()) == false) {
+            throw new DocumentParsingException(context.parser().getTokenLocation(),
+                "Field [%s] is not registered as a %s field type".formatted(fieldName, SemanticTextFieldMapper.CONTENT_TYPE));
+        }
+
+        parseFieldInferenceResultsArray(context, mapperBuilderContext, fieldName);
+    }
+
+    private static void parseFieldInferenceResultsArray(
+        DocumentParserContext context,
+        MapperBuilderContext mapperBuilderContext,
+        String fieldName) throws IOException {
+
+        XContentParser parser = context.parser();
+        NestedObjectMapper nestedObjectMapper = createNestedObjectMapper(context, mapperBuilderContext, fieldName);
+        context.path().add(fieldName); // TODO: Need path manipulation?
```

Comment:
@carlosdelest I don't think we need this path manipulation since we are creating a nested context, can you confirm?

Reply:
I needed that in my tests - the object mapper holds a reference to the context, but we can't rely on the `ObjectMapper` code that updates the context when parsing a nested object, and I needed to do that manually.

Reply:
Interesting. I commented out the path manipulation and my tests work fine... something to investigate further.

Reply:
I trust your tests more than my PoC 😄 . Although we still haven't done the query side of it🤔 .